bundle/jsky.app.ot/src/main/java/jsky/app/ot/tpe/feat/TpeGuidePosFeature.java

```diff
@@ -309,34 +305,43 @@ public void create(TpeMouseEvent tme, TpeImageInfo tii) {
     /**
      * Perform the action of selecting the primary guide star in the group.
      */
-    public void action(TpeMouseEvent tme) {
+    public void action(final TpeMouseEvent tme) {
         if (!getIdentifyPrimary()) return;
 
-        Option<Tuple2<GuideProbe, SPTarget>> res = locatePosition(tme);
-        if (res.isEmpty()) return;
-
-        TargetObsComp   toc = getTargetObsComp();
-        SPTarget     target = res.getValue()._2();
-        PrimaryTargetToggle.instance.toggle(toc, target);
-        _iw.getContext().targets().commit();
+        locatePosition(tme).foreach(tup -> {
+            final SPTarget target   = tup._2();
+            final TargetObsComp toc = getTargetObsComp();
+            PrimaryTargetToggle.instance.toggle(toc, target);
+            _iw.getContext().targets().commit();
+        });
     }
 
 
     /**
      */
-    public boolean erase(TpeMouseEvent tme) {
+    public boolean erase(final TpeMouseEvent tme) {
         final Option<Tuple2<GuideProbe, SPTarget>> res = locatePosition(tme);
-        if (res.isEmpty()) return false;
-
-        final TargetObsComp     toc = getTargetObsComp();
-        final TargetEnvironment env = getTargetEnvironment();
-        final Option<GuideProbeTargets> gtOpt = env.getPrimaryGuideProbeTargets(res.getValue()._1());
-        if (gtOpt.isEmpty()) return false;
-
-        final GuideProbeTargets gt = gtOpt.getValue().removeTarget(res.getValue()._2());
-        toc.setTargetEnvironment(env.putPrimaryGuideProbeTargets(gt));
-        _iw.getContext().targets().commit();
-        return true;
+        return res.map(tup -> {
+            final GuideProbe probe = tup._1();
+            final SPTarget target  = tup._2();
+
+            final TargetObsComp     toc = getTargetObsComp();
+            final TargetEnvironment env = getTargetEnvironment();
+            if (env == null) return false;
+
+            // We do not allow deletion of auto guide stars.
+            final GuideGroup        gp  = env.getOrCreatePrimaryGuideGroup();
+            if (gp.isAutomatic()) return false;
+
+            final Option<GuideProbeTargets> gtOpt = gp.get(probe);
+            gtOpt.map(gt -> {
+                final GuideProbeTargets gtNew = gt.removeTarget(target);
+                toc.setTargetEnvironment(env.putPrimaryGuideProbeTargets(gtNew));
+                _iw.getContext().targets().commit();
+                return true;
+            }).getOrElse(false);
+            return true;
```

Comment:
Yes, you are correct: I should be returning the results of the `gtOpt.map`.
Thanks for catching that.